Analyses manifest backup to local files & Reset analysis manifest & reannotate

SeriousHorncat opened the pull request:
## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] My code follows the style guidelines enforced by static analysis tools.
- [x] If it is a core feature, I have added thorough tests.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [x] New or existing JSON is formatted using JQ
- [x] Will this be part of a product update? If yes, please write one phrase about this update.

<!-- Delete the tasks from the above list that are Not Applicable for your pull request -->

## Pull Request Details

Wrike Ticket - [Draft script to source environment for Rosalution REST API & and  reset analysis manifest to requeue annotations](https://www.wrike.com/open.htm?id=1629378111)

Changes made:

- Replaced the 'annotate-all-analyses.sh' script with a script that queues annotations for a small subset of analyses instead.
- Created a environment source script to set expected Rosalution REST API environment variables for scripts to use for REST API credentials
- Script that takes a docker container that has the Rosalution MongoDB

**To Review:**
<!-- Make a to do list of things to check for to approve the pull request -->
<!-- Modify the below list as appropriate by editing and deleting text that is not applicable-->

- [x] Static Analysis by Reviewer
- [x] Follow instructions from task Wrike Ticket <https://www.wrike.com/open.htm?id=1624357717> to reset manifest and queue annotations. source the environment using `rosalutionrc.sh`.
- [x] All Github Actions checks have passed.

JmScherer commented:
Having seen this script run, it looks and works great on the target system! Attempting to run the script for the review resulted in two main issues:

1. The script does not appear to run at all on MacOS with Zsh. Specifically, the `-v` unary operators aren't valid when run in the zsh environment and results in an error. This has been discussed at length and a fix has been found to make the erroneous if statements POSIX compliant.

2. When trying to run the script suite on the CGDS development cluster, the script would fail silently and complete without an error code. This is because `curl` was failing due to an issue with an invalid SSL cert when trying to request out to `https://dev.cgds.uab.edu`. The fix for this was adding a `-k` flag for the curl operations to allow for insecure connections.

There appeared to be other minor things like how we're checking for heart-beat responses that wasn't caught when `curl` silently failed.

Overall, these scripts should set the standard for how CGDS writes scripts moving forward and likely will be once the issues are fixed!

SeriousHorncat commented:
@JmScherer  

For #1: Found each of the non POSIX compliant checks for variables being set and updated them.  I referred to https://stackoverflow.com/questions/42655304/how-do-i-check-if-a-variable-is-set-in-zsh and https://unix.stackexchange.com/questions/109625/shell-scripting-z-and-n-options-with-if stackexchange discussions. 

For #2: Followed up with each of the cURL calls made in the shell scripts and updated them to include the -k option since we trust ourselves as a source. I also updated the checks for success of running cURL and not just the return result also to better capture success. 

Hopefully my changes cover the issues you discovered!

JmScherer commented on etc/api/rosalutionrc.sh:
```diff
@@ -0,0 +1,48 @@
+
+echo "Visit '$BASE_URL/account' for API account credentials: "
+
+if [[ ! -v INPUT_CLIENT_ID ]]
```
```suggestion
if [ -z "${INPUT_CLIENT_ID+1}" ]
```

JmScherer commented:
Restored latest Rosalution backup from mongo locally

<img width="891" alt="Screenshot 2025-04-23 at 1 05 37 PM" src="https://github.com/user-attachments/assets/98751299-f22f-43d9-8261-c7ec3fbc3394" />

`source ./rosalutionrc.sh`

<img width="765" alt="Screenshot 2025-04-23 at 1 11 43 PM" src="https://github.com/user-attachments/assets/a40331d5-4971-43df-af80-e9dcb824c606" />

Manifest before running reset

![Screenshot 2025-04-23 at 1 25 46 PM](https://github.com/user-attachments/assets/9bf3bade-1862-4d5a-8420-a1b25ce89d39)

`./reset-analysis-manifest-annotate.sh rosalution-rosalution-db-1 "CPAM0002"`

<img width="784" alt="Screenshot 2025-04-23 at 1 32 18 PM" src="https://github.com/user-attachments/assets/b42a417f-70fb-4321-9c77-291ce5497199" />

![Screenshot 2025-04-23 at 1 32 46 PM](https://github.com/user-attachments/assets/f1d48b63-1931-4b5e-9e8b-1b0035000a7d)

![Screenshot 2025-04-23 at 1 31 54 PM](https://github.com/user-attachments/assets/68724852-570d-4fd6-b57d-cdb73decdb63)

JmScherer reviewed (commented):
Looks great and runs well, just needed one change for the rosalutionrc script

JmScherer reviewed (approved):
Works well! Approved
